src/controller/CHIPDeviceController.cpp

```diff
@@ -853,6 +860,18 @@ CHIP_ERROR DeviceCommissioner::PairDevice(NodeId remoteDeviceId, RendezvousParam
         ReturnErrorOnFailure(device->SetCSRNonce(ByteSpan(mCSRNonce)));
     }
 
+    // If the AttestationNonce is passed in, using that else using a random one..
+    if (params.HasAttestationNonce())
+    {
+        ReturnErrorOnFailure(device->SetAttestationNonce(params.GetAttestationNonce().Value()));
+    }
+    else
+    {
+        uint8_t mAttestationNonce[kAttestationNonceLength];
+        Crypto::DRBG_get_bytes(mAttestationNonce, sizeof(mAttestationNonce));
+        ReturnErrorOnFailure(device->SetAttestationNonce(ByteSpan(mAttestationNonce)));
```

Comment:
This here, is, where in instead of storing this in the Device, we should have a DeviceCommissioner member that maps between `Device*` -> `CommissioningContext`, where `CommissioningContext` is a new struct that encapsulates "in flight" information (e.g. the DAC, PAI certs, the nonce, etc) for a given commissioning. This would offload it from Device which is being overloaded.

Reply:
what if we move the DAC and PAI certs to FabricTable/FabricInfo?

Reply:
We cannot move to FabricInfo since FabricInfo is the persistent information about a commissioned fabric. DAC and PAI are transient per-device artifacts of the commissioning flows. We only need to hang on to them between callbacks. Right now the CSR had been put in Device, but it's not really the appropriate location as was stated above by Pankaj/Michael. For now let's leave in Device, but we need to move it to DeviceCommissioner somehow

Reply:
ok